File: MacTemplet/Cagtegory/NSPopover+Helper.m
//
//  NSPopover+Helper.m
//  MacTemplet
//
//  Created by Bin Shang on 2019/6/19.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

#import "NSPopover+Helper.h"

@implementation NSPopover (Helper)

+(instancetype)popoverWithController:(NSViewController *)controller{
    assert([self isSubclassOfClass:NSPopover.class]);
    
    NSPopover *popover = [[NSPopover alloc]init];
    popover.appearance = [NSAppearance appearanceNamed:NSAppearanceNameAqua];
    popover.behavior = NSPopoverBehaviorTransient;
    popover.contentViewController = controller;
    return popover;
}

@end

File: MacTemplet/CocoaExpand/NSPopover+Helper.swift

//
//  NSPopover+Helper.swift
//  MacTemplet
//
//  Created by Bin Shang on 2019/11/22.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

import Cocoa

@objc public extension NSPopover {

    static func create(controller: NSViewController) -> Self {
        let popover: NSPopover = self.init()
        popover.appearance = NSAppearance(named: .aqua)
        popover.behavior = .transient
        popover.contentViewController = controller;
        return popover as! Self;
    }
}
